genson/src/main/java/com/owlike/genson/convert/DefaultConverters.java

```diff
@@ -1337,4 +1337,150 @@ public T deserialize(ObjectReader reader, Context ctx) throws Exception {
       return value;
     }
   }
+
+  public static class OptionalConverterFactory implements Factory<Converter<Optional<?>>> {
+    public static final OptionalConverterFactory instance = new OptionalConverterFactory();
+    private OptionalConverterFactory() {
+    }
+
+    @Override
+    public Converter<Optional<?>> create(final Type type, final Genson genson) {
+      Type optionalType = typeOf(0, type);
+      //noinspection unchecked
+      return new OptionalConverter(genson.provideConverter(optionalType));
+    }
+  }
+
+  @HandleClassMetadata
+  @HandleBeanView
+  public static class OptionalConverter<T> implements Converter<Optional<T>> {
+
+    private Converter<T> converter;
+
+    public OptionalConverter(Converter<T> converter) {
+      this.converter = converter;
+    }
+
+    @Override
+    public void serialize(final Optional<T> optional, final ObjectWriter writer, final Context ctx) throws Exception {
+      writer.beginObject();
+      writer.writeMetadata("class", Optional.class.getName());
```

Comment:
I'm not sure we should be writing `@class` attribute directly -- it will be written out by the `ClassMetadataConverter` if necessary, based on configuration. If Genson can derive that somethings is an `Optional` based on the static type of a property, then it doesn't really have to be written out anyway.

I would remove it from here and get rid of the `@ HandleClassMetadata` above, and let `ClassMetadataConverter` do its job.

And yes, we should rethink whether we should do it ourselves within the `MapConverter` as well, although that converter is outside of Genson core, so I'm not as worried about it.